emukit/core/loop/stopping_conditions.py

```diff
@@ -31,4 +37,31 @@ def should_stop(self, loop_state: LoopState) -> bool:
         :param loop_state: Object that contains current state of the loop
         :return: True if maximum number of iterations has been reached
         """
-        return loop_state.iteration >= self.i_max
+        status = loop_state.iteration >= self.i_max
+        if status is True:
+            _log.info("Stopped after {} Evaluations".format(self.i_max))
+        return status
+
+
+class ConvergenceStoppingCondition(StoppingCondition):
+    """ Stops once we choose a point within eps of a previous
+    point (with respect to euclidean norm), a notion of convergence"""
+    def __init__(self, eps: float) -> None:
+        """
+        :param eps: minimum distance between
+        two consecutive x's to keep running the model
+        """
+        self.eps = eps
+
+    def should_stop(self, loop_state: LoopState) -> bool:
+        """
+        :param loop_state: Object that contains current state of the loop
+        :return: True if consecutive x's are too close
+        """
+        if loop_state.iteration < 2:
+            # less than 2 evaluations so cannot calculate distance
+            return False
+        status = math.sqrt(sum((loop_state.X[-1, :] - loop_state.X[-2, :]) ** 2)) <= self.eps
```

Comment:
anything wrong with `numpy.linalg.norm`? also, same point about boolean variable name

Reply:
The problem with using np here, is that you end up with a np boolean object. This doesn't work with the assert statements in the tests.

I kept getting assert error False is False
 as one False was the numpy bool and one was the standard python bool

Reply:
not sure i am following. `linalg.norm` can certainly return a single float, and `float <= float` returns python boolean. what am i missing?

Reply:
np.linalg.norm returns a numpy float. So the comparison provides a numpy bool

Reply:
oh, ok, missed that, thanks. in that case how about just converting numpy float to python float?